server/src/main/java/org/elasticsearch/cluster/coordination/CoordinationDiagnosticsService.java

```diff
@@ -967,16 +991,20 @@ private <R extends ActionResponse, T> Scheduler.Cancellable sendTransportRequest
                 logger.trace("Opened connection to {}, making transport request", masterEligibleNode);
                 // If we don't get a response in 10 seconds that is a failure worth capturing on its own:
                 final TimeValue transportTimeout = TimeValue.timeValueSeconds(10);
-                transportService.sendRequest(
-                    masterEligibleNode,
-                    transportActionType.name(),
-                    transportActionRequest,
-                    TransportRequestOptions.timeout(transportTimeout),
-                    new ActionListenerResponseHandler<>(
-                        ActionListener.runBefore(fetchRemoteResultListener, () -> Releasables.close(releasable)),
-                        transportActionType.getResponseReader()
-                    )
-                );
+                try {
+                    transportService.sendRequest(
+                        masterEligibleNode,
+                        transportActionType.name(),
+                        transportActionRequest,
+                        TransportRequestOptions.timeout(transportTimeout),
+                        new ActionListenerResponseHandler<>(
+                            ActionListener.runBefore(fetchRemoteResultListener, () -> Releasables.close(releasable)),
+                            transportActionType.getResponseReader()
+                        )
+                    );
+                } catch (Exception e) {
+                    responseConsumer.accept(responseTransformationFunction.apply(null, e));
+                }
```

Comment:
We only catch a `NodeNotConnectedException` thrown from `ConnectionManager#getConnection` so in theory something else could escape from here, although in practice that's the only exception that any implementations will throw. We should probably just `catch Exception` here to make sure, rather than putting the onus on callers. Would you open an issue for the distrib team to fix that?

Reply:
Thanks David. Opened https://github.com/elastic/elasticsearch/issues/89274